File: 55. Jump Game/solution.py
class Solution:
    def canJump(self, nums: list[int]) -> bool:
        if len(nums) == 1:
            return True
        dp = [0 for i in range(len(nums))]
        dp[0] = 1 if nums[0] else 0
        for i in range(len(nums)):
            if dp[-1]:
                return True
            if dp[i]:
                for j in range(1, nums[i] + 1):
                    if i + j < len(nums):
                        dp[i + j] = 1
        return dp[-1]

File: 55. Jump Game/solution.go
package main

func canJump(nums []int) bool {
	if len(nums) == 1 {
		return true
	}
	lastPosition := nums[0]
	for i := 0; i < lastPosition; i++ {
		if i+nums[i]+1 > lastPosition {
			lastPosition = i + nums[i] + 1
		}
		if i == len(nums)-1 {
			return true
		}
	}
	return false
}

func main() {
	nums := []int{2, 3, 1, 1, 4}
	canJump(nums)
}
